packages/dataviews/src/dataviews-layouts/table/index.tsx

```diff
@@ -102,15 +110,40 @@ function TableColumnField< Item >( {
 	primaryField,
 	item,
 	field,
+	isItemClickable,
+	onClickItem,
 }: TableColumnFieldProps< Item > ) {
+	const isPrimaryField = primaryField?.id === field.id;
+	const isItemClickableField = isItemClickable( item ) && isPrimaryField;
 	return (
 		<div
 			className={ clsx( 'dataviews-view-table__cell-content-wrapper', {
-				'dataviews-view-table__primary-field':
-					primaryField?.id === field.id,
+				'dataviews-view-table__primary-field': isPrimaryField,
 			} ) }
 		>
-			<field.render { ...{ item } } />
+			<div
+				className={ clsx( 'dataviews-view-table__cell-content', {
+					'dataviews-view-table__cell-content--clickable':
+						isItemClickableField,
+				} ) }
+				tabIndex={ isItemClickableField ? 0 : undefined }
+				role="button"
+				onClick={ () => {
+					if ( isItemClickableField ) {
+						onClickItem( item );
+					}
+				} }
+				onKeyDown={ ( event ) => {
+					if (
+						( event.key === 'Enter' || event.key === '' ) &&
+						isItemClickableField
+					) {
+						onClickItem( item );
+					}
+				} }
```

Comment:
It feels like we're repeating the same logic in all the places where we want to make something clickable. Maybe we can do have a reusable useClickableItemProps hook and reuse it in all these places:

```
const clickableProps = useClickableItemProps( item ); 
```

Reply:
Great idea! Thanks! 🙇 507d60b0db8dc4c056e3699a567394025d352d98